### PR TITLE
Add L2 test specification and implementation

### PR DESCRIPTION
**Pull request by chaithents**

### CODE REVIEW

**Comment by Ulrond** [comment]
can you share the output of the tests, so we can review it..

-> It must run completely without crashing on x86
-> It must run and can you give a report of the testing suite on arm target.

**Comment by Ulrond** [comment]
this is a fork of the repo and not a branch off the actual repo, please re-do it to be a branch see -> https://github.com/rdkcentral/rdk-halif-test-rmf_audio_capture/blob/main/CONTRIBUTING.md

**Review by Ulrond — CHANGES_REQUESTED** [review]
blocked on this being a fork.

please delete this fork and create a branch

File: docs/pages/L2_TestSpecification.md [hunk 1]
```diff
@@ -15,38 +15,37 @@
...
 - `Amlogic` \- `SoC` manufacturer <https://en.wikipedia.org/wiki/Amlogic>
+- `Realtek` \- `SoC` manufacturer <https://en.wikipedia.org/wiki/Realtek>
 - `Soc Vendor` \- Definition to encompass multiple vendors
 - `Unit Tests` \- C Function tests that run on the target hardware
 - `Common Testing Framework` \- Off the shelf 3rd Party Testing Framework, or framework that does not require infrastructure to control it. That's not to say it cannot be controlled via infrastructure if required. Examples of which are.
   - `GTest` \- Google Test Suit <https://google.github.io/googletest>
   - `CUnit` \- C Testing Suit <http://cunit.sourceforge.net/>
   - `Unity` \- C Embedded Testing Suit <http://www.throwtheswitch.org/unity>
-  - `ut-core` \- Common Testing Framework <https://github.com/rdkcentral/ut-core>, which wraps a open-source framework that can be expanded to the requirements for future framework.
+  - `ut-core` \- Common Testing Framework <https://github.com/comcast-sky/rdk-components-ut-core>, which wraps a open-source framework that can be expanded to the requirements for future comcast-sky framework.
```
Comment (Ulrond) [inline docs/pages/L2_TestSpecification.md:48]: these are all wrong, rdkcentral/ut-core was correct

**Comment by Ulrond** [comment]
Your testing requirements conflict with here -> https://github.com/rdkcentral/rdk-halif-test-rmf_audio_capture/pull/4/files#diff-cfbf2cf5c46f28c9994f07c4d9430995ba31f357d54ce87b92180ddd4780adbf

**Comment by Ulrond** [comment]
> Your testing requirements conflict with here -> https://github.com/rdkcentral/rdk-halif-test-rmf_audio_capture/pull/4/files#diff-cfbf2cf5c46f28c9994f07c4d9430995ba31f357d54ce87b92180ddd4780adbf

> Let me discuss internally and see what the best approach maybe, could be merge you changes first, then rebase on the document updates.

**Comment by srinivasgtl** [comment]
All L2 updates available in #4

**Pull request closed without merging**